devtools/x/src/tools.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{context::XContext, installer, Result};
+use log::info;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(long)]
+    /// Run in 'check' mode. Exits with 0 if all tools installed. Exits with 1 and if not, printing failed
+    check: bool,
+}
+
+pub fn run(args: Args, xctx: XContext) -> Result<()> {
+    let tools = xctx.config().tools();
+    let mut failed: bool = false;
+    for (key, value) in tools {
+        if !args.check {
+            installer::install_if_needed_or_die(key, value);
```

Comment:
Why or die? might be better to have this return an error and handle the exit code in 1 single place (like you are doing below)

Reply:
The plan is to use the installer in multiple locations, I suppose I can try to find a good one liner to wrap the calls when used in xtext/xbuild.    The `x tools` is mostly about pre-installation in CI environments, and if a dev wants to setup for offline work.

Reply:
removed, the or_die bit.   will need a good one line when used in xtest/xbuild.